src/axom/core/Array.hpp

```diff
@@ -57,9 +93,9 @@ constexpr IndexType ZERO = 0;
  *  * <b> External Storage </b> <br />
  *
```

Comment:
Here or in MCArray, please compare and contrast the two classes.  Why two?  How are they different, and what are they used for?  These two classes mayor may not be temporary, but as long as we have two classes as close as these we need documentation on why they're different.